cli/azd/pkg/azure/arm_template.go

```diff
@@ -20,6 +28,22 @@ type ArmTemplate struct {
 	Outputs        ArmTemplateOutputs              `json:"outputs"`
 }
 
+func (t ArmTemplate) TargetScope() DeploymentScope {
+	u, err := url.Parse(t.Schema)
+	if err != nil {
+		panic(fmt.Sprintf("bad schema for template: %s", t.Schema))
```

Comment:
For slight correctness: would you consider if we return `err` here, and `panic` inside the `bicep_provider`? It's not clear from this place whether the `ArmTemplate` is expected to always be valid.

In `bicep_provider`, the idea emerges that we compiled the bicep, and thus bicep would guarantee that we got some valid file.

Reply:
Happy to do this, and love the line of thinking here.

Reply:
Ended up just refactoring this so that in addition to this being error returning, we also don't panic in the bicep provider, to provide a slightly better experience for the user who brings a template with an unsupported scope.